### ISSUE TITLE
Anti direct connect bypass + Ping servers through proxy

### ISSUE THREAD

**Issue by pawa2401** [MeteorDevelopment/meteor-client#4116]
### Describe the feature

Please make a workaround for this:
https://www.spigotmc.org/threads/tool-anti-direct-connect.40938/
It seems that more and more smaller servers have this feature (or a similar one). It makes the server list full of trash after a while and it's really annoying.

Also if it wasn't a thing already, please make so that the client pings the servers on the server list through the proxy that is selected.
(I'm not sure if this is implemented already, but when I used a really high latency proxy, the server info seemed to update with the same speed)

### Before submitting a suggestion

- [X] This feature doesn't already exist in the client. (I have checked every module and their settings on the **latest dev build**)

- [X] This wasn't already suggested. (I have searched suggestions on GitHub and read the FAQ)

**Comment by Paddyk45** [comment]
That should be easy, can you send a server that has this plugin?

**Comment by Paddyk45** [comment]
And I tested, Server Pings are proxied

**Comment by pawa2401** [comment]
I don't know if this server has this one in particular, but my friend wants me to play "clearmc.eu", and it has something similar.

**Comment by Paddyk45** [comment]
I made a small mod, it's not the best but it works (at least when I tested it) you can download it here: https://github.com/Paddyk45/Anti-AntiDirectConnect/releases

**Comment by pawa2401** [comment]
Thanks for the effort, but unfortunately it doesn't work for me. Have you tested it with a proxy?

**Comment by Paddyk45** [comment]
> Thanks for the effort, but unfortunately it doesn't work for me.
What server?

**Comment by pawa2401** [comment]
> What server?
clearmc.eu
The one I mentioned earlier

**Comment by Paddyk45** [comment]
I tried it on that one and it didn't have any direct connect kick

**Comment by Paddyk45** [comment]
Are you playing on a cracked client?

**Comment by Paddyk45** [comment]
Also, the mod should work after the second connect if it didn't work after the first one

**Comment by pawa2401** [comment]
I have a cracked account selected and my game is running in offline mode
I'd be nice if I didn't have to use my premium account though. Most of the servers I play on are cracked so I don't use it often.

Also I've tried reconnecting a bunch of times and it never lets me through, no matter how much I try.

**Comment by Paddyk45** [comment]
> I have a cracked account selected and my game is running in offline mode I'd be nice if I didn't have to use my premium account though. Most of the servers I play on are cracked so I don't use it often.
> 
> Also I've tried reconnecting a bunch of times and it never lets me through, no matter how much I try.

And you did check "Bypass AntiDirectConnect"?

**Comment by pawa2401** [comment]
Yes, I have it checked, and It finally let me through the first time

**Comment by pawa2401** [comment]
Could this be implemented to Meteor by default after some reworks? This doesn't seem to work too reliably, but It at least works.

**Comment by Paddyk45** [comment]
> Could this be implemented to Meteor by default after some reworks? This doesn't seem to work too reliably, but It at least works.

I could open a PR, but i doubt it will get merged

**Comment by Paddyk45** [comment]
Done
